compiler/src/iree/compiler/Codegen/Common/CombineLayoutTransformation.cpp

```diff
@@ -383,25 +383,50 @@ combineRelayoutOpChain(RewriterBase &rewriter, MapScatterOp mapScatterOp,
   }
 }
 
-static MapScatterOp
-insertIdentityMapScatter(RewriterBase &rewriter,
-                         IREE::Codegen::StoreToBufferOp storeOp) {
-  Location loc = storeOp->getLoc();
+// Insert identity map_scatter op for parallel_insert_slice op and store op.
+// Return Null for any other unsupported op.
+static MapScatterOp insertIdentityMapScatter(RewriterBase &rewriter,
+                                             Operation *op) {
+  Location loc = op->getLoc();
   OpBuilder::InsertionGuard g(rewriter);
-  rewriter.setInsertionPoint(storeOp);
-  auto mapScatterDest =
-      rewriter
-          .create<tensor::EmptyOp>(
-              loc, memref::getMixedSizes(rewriter, loc, storeOp.getBuffer()),
-              storeOp.getTensor().getType().getElementType())
-          .getResult();
-  auto mapScatterOp = MapScatterOp::createIdentityMapScatter(
-      rewriter, loc, storeOp.getTensor(), mapScatterDest);
-  rewriter.modifyOpInPlace(storeOp, [&]() {
-    storeOp.getTensorMutable().assign(mapScatterOp.getResult(0));
-  });
-  LDBG("Created identity map_scatter:\n" << mapScatterOp);
-  return mapScatterOp;
+  if (auto storeOp = dyn_cast<IREE::Codegen::StoreToBufferOp>(op)) {
+    rewriter.setInsertionPoint(storeOp);
+    auto mapScatterDest =
+        rewriter
+            .create<tensor::EmptyOp>(
+                loc, memref::getMixedSizes(rewriter, loc, storeOp.getBuffer()),
+                storeOp.getTensor().getType().getElementType())
+            .getResult();
+    auto mapScatterOp = MapScatterOp::createIdentityMapScatter(
+        rewriter, loc, storeOp.getTensor(), mapScatterDest);
+    rewriter.modifyOpInPlace(storeOp, [&]() {
+      storeOp.getTensorMutable().assign(mapScatterOp.getResult(0));
+    });
+    LDBG("Created identity map_scatter:\n" << mapScatterOp);
+    return mapScatterOp;
+  } else if (auto parallelinsertsliceop =
+                 dyn_cast<tensor::ParallelInsertSliceOp>(op)) {
+    rewriter.setInsertionPoint(parallelinsertsliceop->getParentOp());
+    Value source = parallelinsertsliceop.getSource();
+    auto mapScatterDest =
+        rewriter
+            .create<tensor::EmptyOp>(
+                loc, tensor::getMixedSizes(rewriter, loc, source),
+                dyn_cast<mlir::TensorType>(source.getType()).getElementType())
+            .getResult();
+    auto mapScatterOp = MapScatterOp::createIdentityMapScatter(
+        rewriter, loc, source, mapScatterDest);
+    rewriter.modifyOpInPlace(parallelinsertsliceop, [&]() {
+      parallelinsertsliceop.getSourceMutable().assign(
+          mapScatterOp.getResult(0));
+    });
+    LDBG("Created identity map_scatter:\n" << mapScatterOp);
+    return mapScatterOp;
+  } else {
+    op->emitError("Unsupported Op, identity map_scatter is created only for "
+                  "storeOp & parallel_insert_slice.\n");
+    return nullptr;
+  }
```

Comment:
nit: This can be simplified by just using `op->getOperand(0)` and `op->setOperand(0, mapScatterOp.getResult(0))`. Something like:
```suggestion
    assert(isa<IREE::Codegen::StoreToBufferOp,
               tensor::ParallelInsertSliceOp>(op) &&
           "expected store_to_buffer or parallel_insert_slice op");
    rewriter.setInsertionPoint(op);
    auto destType = cast<RankedTensorType>(op->getOperand(1).getType());
    Value mapScatterDest = rewriter.create<tensor::EmptyOp>(
        loc, memref::getMixedSizes(rewriter, loc, op->getOperand(1)),
        destType.getElementType());
    auto mapScatterOp = MapScatterOp::createIdentityMapScatter(
        rewriter, loc, op->getOperand(0), mapScatterDest);
    rewriter.modifyOpInPlace(op, [&]() {
      op->setOperand(0, mapScatterOp.getResult(0));
    });
    LDBG("Created identity map_scatter:\n" << mapScatterOp);
    return mapScatterOp;
```